### DISCUSSION TITLE
Rust version of jax

### DISCUSSION THREAD

**Discussion by any35** [jax-ml/jax#9048] [Ideas]
Hello, I want to know if there are plans to implement rust version of jax? thans.

**Comment by mattjj** [comment]
Thanks for the question! Actually, this came up in #5938, so I'll repeat an interesting factoid: @froystig [used to work on Rust](https://github.com/rust-lang/rust//commits?author=froystig), more than a decade ago! And while doing that at Mozilla he met another JAX cofounder @cdleary, who I think was working on JavaScript JITs at the time.

Having a Rust version sounds like a cool idea, but it's out of scope for us right now.

**Comment by jgsimard** [comment]
@mattjj  Hi, is it in scope now or in the future?

**Reply by mattjj** [reply]
Not in scope in the foreseeable future, sorry.

**Comment by Oceania2018** [comment]
What about expose the C interface for other languages to bind?

**Comment by rjzak** [comment]
I too would be interested in Rust or C for Jax, so that Jax models could be used outside of Python.

**Comment by cksac** [comment]
Hi, I am currently working on implement a ML framework similar to JAX in Rust, https://github.com/cksac/rai. 
Still in early stage, it now use Candle as backend, I am planning to add XLA/IREE backend later

**Reply by rjzak** [reply]
Starred, thanks!
